nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2449,24 +2609,96 @@ async fn disk_delete(
     let apictx = rqctx.context();
     let nexus = &apictx.nexus;
     let path = path_params.into_inner();
-    let organization_name = &path.organization_name;
-    let project_name = &path.project_name;
-    let disk_name = &path.disk_name;
+    let disk_selector = params::DiskSelector::new(
+        Some(path.organization_name.into()),
+        Some(path.project_name.into()),
+        path.disk_name.into(),
+    );
     let handler = async {
         let opctx = OpContext::for_external_api(&rqctx).await?;
-        nexus
-            .project_delete_disk(
-                &opctx,
-                &organization_name,
-                &project_name,
-                &disk_name,
-            )
-            .await?;
+        let disk_lookup = nexus.disk_lookup(&opctx, &disk_selector)?;
+        nexus.project_delete_disk(&disk_lookup).await?;
         Ok(HttpResponseDeleted())
     };
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Attach a disk to an instance
+#[endpoint {
+    method = POST,
+    path = "/v1/disks/{disk}/attach",
+    tags = ["disks"],
+}]
+async fn disk_attach_v1(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<params::DiskPath>,
+    query_params: Query<params::OptionalProjectSelector>,
+    body: TypedBody<params::InstanceIdentifier>,
+) -> Result<HttpResponseAccepted<Disk>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let query = query_params.into_inner();
+    let body = body.into_inner();
+    let instance_selector = params::InstanceSelector {
+        instance: body.instance,
+        project_selector: query.project_selector.clone(),
+    };
+    let disk_selector = params::DiskSelector {
+        disk: path.disk,
+        project_selector: query.project_selector,
+    };
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let instance_lookup =
+            nexus.instance_lookup(&opctx, &instance_selector)?;
+        let disk_lookup = nexus.disk_lookup(&opctx, &disk_selector)?;
+        let disk = nexus
+            .instance_attach_disk(&opctx, &instance_lookup, &disk_lookup)
+            .await?;
```

Comment:
Note to self: This won't work. The lookup is used for the thing in the path (the disk in this case) not the thing in the body (instance).